lib/Hakyll/Core/Compiler/Internal.hs

```diff
@@ -90,11 +94,25 @@ instance Monoid CompilerWrite where
 
 
 --------------------------------------------------------------------------------
-data CompilerResult a where
-    CompilerDone     :: a -> CompilerWrite -> CompilerResult a
-    CompilerSnapshot :: Snapshot -> Compiler a -> CompilerResult a
-    CompilerError    :: [String] -> CompilerResult a
-    CompilerRequire  :: (Identifier, Snapshot) -> Compiler a -> CompilerResult a
+data Reason a
+    -- | An exception occured during compilation
+    = CompilationFailure a
+    -- | Absence of any result, most notably in template contexts
+    | NoCompilationResult a
+
+
+-- | Unwrap a `Reason`
```

Comment:
I guess `unReason` might have followed the naming conventions, but it seemed kinda unreasonable :-D

Reply:
:grin: